### ISSUE TITLE
Documentation: Structural inconsistencies

### ISSUE THREAD

**Issue by ufrshubham** [flame-engine/flame#2003]
# What could be improved

Flame's documentation covers almost everything that it has to offer, but at times it feels out of order. For example:

1. `Router` section is placed way before `Components` section, but the documentation in `Router` assumes that users already knows what `FlameGame`, `Component`s, `FCS` and `overlay` is. Probably `Router` should be moved below `Components` section. 
![Router reorder](https://user-images.githubusercontent.com/33748002/193562676-cd352f51-b1e4-425b-a921-a8bbb0f04a5b.png)

2. `Game loop` section directly starts with explaining `FlameGame` class. The actual game loop is explained quite late in that section. Since Flame is essentially a flutter package, majority of the audience reading the docs is (and probably will be) app developers. For many of them, the core game-loop (infinite update-render cycle) is quite new. It would be good to introduce the concept first and then the flame specific implementation (`FlameGame` & `Game`).
![image](https://user-images.githubusercontent.com/33748002/193562964-be626867-15c9-466e-89c1-bea2f47f305e.png)

3. This one is a bit debatable, but the `Components` section feels like a giant compared to other pages. Probably it would be a good idea to split it into multiple sub-topics. Similar to how it is done for `Inputs` and `Rendering`.
![image](https://user-images.githubusercontent.com/33748002/193563496-b987fd6d-c039-4738-ac34-dafdcbedf39b.png)

4. `Ray casting and Ray tracing` probably deserves its own section altogether. Right now it lives under `Collision detection`.

# Good to know
Make sure that you comment on the issue that you will be working on this before starting to program.

To set up the Flame environment you need to follow the instructions in our [contributing.md](https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md#environment-setup) doc, if you have any problems setting up your environment (or other issues of course) feel free to ask for help in the Flame channel on our [Discord server](https://discord.com/invite/pxrBmy4).

Since this is a documentation change, you need to be able to build the docs. You can do that by running melos sphinxdoc after you have followed the instructions in the contributing.md doc and also [installed Sphinx on your system](https://www.sphinx-doc.org/en/master/usage/installation.html). After you have built the docs you can open the doc/_build/html/index.html file in your web browser to see the changes.

Make sure to include a video of the example when you upload the PR so that it will be easier for us to review it. (You can attach videos directly in the PR, so no need to upload them somewhere else)

**Happy hacking!**

**Comment by AbhishekTiwari23** [comment]
I want to contribute to this. Please assign it to me

**Comment by AbhishekTiwari23** [comment]
Thanks for assigning me. I will respond as soon as possible.

**Comment by omar6260** [comment]
I want to contribute to this. Please assign it to me

**Comment by ufrshubham** [comment]
@AbhishekTiwari23 feel free to open a draft PR if you want to have an early review or discussion

**Comment by ufrshubham** [comment]
> I want to contribute to this. Please assign it to me

It is already assigned to someone, but if you want you can take a look at the other unassigned documentation issue.

**Comment by munsterlander** [comment]
@AbhishekTiwari23 Let me know if you plan to abandon this PR.  I have some additional changes stacked up that I am waiting for your PR before I submit as I don't want to deal with the merge conflicts.

**Comment by AbhishekTiwari23** [comment]
> @AbhishekTiwari23 Let me know if you plan to abandon this PR. I have some additional changes stacked up that I am waiting for your PR before I submit as I don't want to deal with the merge conflicts.

I am working on it. If You want any change, then please let me know.

**Comment by ufrshubham** [comment]
@AbhishekTiwari23 no pressure, but just checking in if you are still planning to work on this and/or facing any problem with this?

**Comment by AbhishekTiwari23** [comment]
> @AbhishekTiwari23 no pressure, but just checking in if you are still planning to work on this and/or facing any problem with this?

I will make a PR in 24 hr the college has started, and I was getting less time for all this, and I am also trying to solve the issues I made in previous days. Please give me the last 24hr. If I cannot make a pr by that time, you can assign it to someone else.
Thanks.

**Comment by munsterlander** [comment]
I have this addressed if they unassign themselves, otherwise, I will resolve the conflicts as they arise.

**Comment by munsterlander** [comment]
No no, by all means, you should try, I am just saying if you decide to back out, then I have it as well.

**Comment by AbhishekTiwari23** [comment]
> No no, by all means, you should try, I am just saying if you decide to back out, then I have it as well.

no issue, I am a beginner and this is my first hacktoberfest. And I am learning.
